cli/command/container/opts.go

```diff
@@ -382,7 +382,7 @@ func parse(flags *pflag.FlagSet, copts *containerOptions, serverOS string) (*con
 
 			if parsed.Type == string(mounttypes.TypeBind) {
 				if hostPart, targetPath, ok := strings.Cut(bind, ":"); ok {
-					if strings.HasPrefix(hostPart, "."+string(filepath.Separator)) || hostPart == "." {
+					if !filepath.IsAbs(hostPart) && strings.HasPrefix(hostPart, ".") {
```

Comment:
I'm not sure we need to match `.` prefix. If there's a rationale behind that, could you expand on that please? 🙂 Also note that, this prefix matching will include hidden files too.

I think this should be just fine:

```suggestion
					if !filepath.IsAbs(hostPart) {
```

Reply:
AFAIK 
> This prefix (`.`) is always required for source since the `-v/--volume/--mount` flags also support setting named volumes.

Reply:
Oh, indeed! 🙈 And looking at the constraints for volume names, we don't allow them to start with a dot. So all good I guess!

```
$ docker volume create .a 
Error response from daemon: create .a: ".a" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path
```